subprojects/architecture-test/src/test/java/org/gradle/architecture/test/PublicApiCorrectnessTest.java

```diff
@@ -70,7 +71,11 @@ public class PublicApiCorrectnessTest {
                 .or(type(BigDecimal.class))
                 .or(type(Element.class))
                 .or(type(QName.class))
-                .as("built-in JDK classes"))
+                .and(
+                    DescribedPredicate.not(resideInAnyPackage("java.util.function"))
+                        .or(type(BiFunction.class))
+                )
+                .as("built-in JDK classes (excluding java.util.function)"))
```

Comment:
```suggestion
                .as("built-in JDK classes"))
```

💅 I'd keep the name here.